### DISCUSSION TITLE
How to use modal from StimulusSlimSelect

### DISCUSSION THREAD

**Discussion by heraldofsolace** [excid3/tailwindcss-stimulus-components#77] [General]
I'm using [StimulusSlimSelect](https://github.com/excid3/stimulus-slimselect) controller and I want to invoke the modal when the user adds a new option. I already have the default modal in the page. I tried giving my SlimSelectController to a shared parent of both the modal and the select and call `this.application.getControllerForElementAndIdentifier(self.modalTarget, "modal").open()` inside the `addable` function of `SlimSelect` but it doesn't work since `open` requires an event.

**Comment by excid3** [comment]
It's easier if you communicate between Stimulus controllers with events. For example, you should be able to add an action to listen for an "open" event and then fire an event to trigger it.

```html
<div data-controller="modal" data-action="open->modal#open">
```

```javascript
const event = new Event("open")
modalElement.dispatchEvent(event)
```

Something roughly like the above should work, but might need some tweaks as I haven't tested it. 👍

**Reply by heraldofsolace** [reply]
Thanks. It works
